### PR TITLE
fix(core): don't allow coinbase transactions in mempool

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---
Rejects coinbase transactions from the mempool

Motivation and Context
---
Coinbase transactions should never be allowed in the mempool

How Has This Been Tested?
---
New unit test

### CODE REVIEW

**Review by SWvheerden — DISMISSED** [review]
LGTM
Tested by syncing a new node, sending a transaction to aurora and back.

**Review by therustmonk — DISMISSED** [review]

File: base_layer/core/src/validation/error.rs [hunk 1]
```diff
@@ -114,6 +114,8 @@ pub enum ValidationError {
     CovenantError(#[from] CovenantError),
     #[error("Invalid or unsupported blockchain version {version}")]
     InvalidBlockchainVersion { version: u16 },
+    #[error("Output in transaction is marked as a coinbase")]
+    MempoolTransactionContainsCoinbase,
```
Comment (CjS77) [inline base_layer/core/src/validation/error.rs]:
```suggestion
    #[error("Standard transaction contains coinbase output")]
    ErroneousCoinbaseOutput
```

Comment (CjS77) [reply base_layer/core/src/validation/error.rs]: This is not necessarily just a mempool thing

**Comment by CjS77** [comment]
IIRC this is the same check that nodes will make before deciding whether to rebroadcast a transaction.
If so, then that's the slight more important instance where this should get thrown out, and why I proposed renaming the error slightly.

Otherwise, I'm OK with the concept.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**